extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/providers/KnownOidcProviders.java

```diff
@@ -19,13 +19,34 @@ public static OidcTenantConfig provider(OidcTenantConfig.Provider provider) {
             case LINKEDIN -> linkedIn();
             case MASTODON -> mastodon();
             case MICROSOFT -> microsoft();
+            case SLACK -> slack();
             case SPOTIFY -> spotify();
             case STRAVA -> strava();
             case TWITCH -> twitch();
             case TWITTER, X -> twitter();
         };
     }
 
+    private static OidcTenantConfig slack() {
+        OidcTenantConfig ret = new OidcTenantConfig();
+        ret.setAuthServerUrl("https://slack.com");
+        ret.setApplicationType(OidcTenantConfig.ApplicationType.WEB_APP);
+        ret.getCredentials().getClientSecret().setMethod(Method.POST);
+
+        var token = ret.getToken();
+        token.setPrincipalClaim("name");
+        token.setSubjectRequired(true);
```

Comment:
That should not be required by default, it is a unique identifier, and is important when you need to distinguish between, for ex, Alice authenticating to more than one provider

Reply:
Slack always prodies it, so I though it is safer it require it, I'll drop it, though I really don't see any harm here?